Source/WebCore/css/calc/CSSCalcOperationNode.cpp

```diff
@@ -1369,6 +1370,14 @@ double CSSCalcOperationNode::evaluateOperator(CalcOperator op, const Vector<doub
     case CalcOperator::Tan: {
         if (children.size() != 1)
             return std::numeric_limits<double>::quiet_NaN();
+        double x = std::fmod(children[0], piDouble * 2);
+        // std::fmod can return negative values.
+        x = x < 0 ? piDouble * 2 + x : x;
+        ASSERT(x >= 0 && x <= piDouble * 2 || std::isnan(x));
+        if (x == piOverTwoDouble)
```

Comment:
Another way to do this is to check for the additional negative value instead of doing addition to normalize to a positive value.

Reply:
Seems cleaner, I agree.

Reply:
I think it's slightly to reason over the smallest range of angles.